logstash-core/src/main/java/org/logstash/config/ir/graph/Graph.java

```diff
@@ -269,7 +269,7 @@ public Integer rank(Vertex vertex) {
     }
 
     public Map<String, List<Vertex>> verticesByHash() {
-        return this.vertices().collect(Collectors.groupingBy(Vertex::uniqueHash));
+        return this.vertices().parallel().collect(Collectors.groupingBy(Vertex::uniqueHash));
```

Comment:
when I benchmarked this during dev it yielded an improvement .

Reading through https://docs.oracle.com/javase/tutorial/collections/streams/parallelism.html seems to confirm that this should in fact work. `.stream().parallel()` can modify a collection's stream to behave as `.parallelStream` would.